server/channels/jobs/s3_path_migration/s3_path_migration.go

```diff
@@ -0,0 +1,276 @@
+			return
+		}
+		totalCount = int(total)
+	}
+
+	// Check if there is metadata for that job.
+	// If there isn't, it will be empty by default, which is the right value.
+	offset := job.Data["start_file_id"]
+
+	doneCount, appErr := worker.getJobMetadata(job, "done_file_count")
+	if appErr != nil {
+		mlog.Error("S3PathMigrationWorker: failed to get done file count", mlog.String("worker", worker.name), mlog.String("job_id", job.Id), mlog.Err(appErr))
+		worker.setJobError(job, appErr)
+		return
+	}
+
+	createAtOffset, appErr := worker.getJobMetadata(job, "start_create_at")
+	if appErr != nil {
+		mlog.Error("S3PathMigrationWorker: failed to get start create_at", mlog.String("worker", worker.name), mlog.String("job_id", job.Id), mlog.Err(appErr))
+		worker.setJobError(job, appErr)
+		return
+	}
+	if createAtOffset == 0 {
+		// Time of the commit because we know no files older than that are affected.
+		// Exact commit was done on 09:54AM June 27, IST.
+		// We take June 26 as an approximation.
+		createAtOffset = int(time.Date(2023, time.June, 26, 0, 0, 0, 0, time.UTC).UnixMilli())
+	}
+
+	const pageSize = 100
+
+	for {
+		select {
+		case <-worker.stop:
+			mlog.Info("Worker: S3 Migration has been canceled via Worker Stop. Setting the job back to pending.",
+				mlog.String("workername", worker.name),
+				mlog.String("job_id", job.Id))
+			if err := worker.jobServer.SetJobPending(job); err != nil {
+				mlog.Error("Worker: Failed to mark job as pending",
+					mlog.String("workername", worker.name),
+					mlog.String("job_id", job.Id),
+					mlog.Err(err))
+			}
+			return
+		case <-time.After(timeBetweenBatches):
+			var files []*model.FileForIndexing
+			tries := 0
+			for files == nil {
+				var err error
+				// Iterate through the rows in fileInfo, starting from the metadata ID
+				files, err = worker.store.FileInfo().GetFilesBatchForIndexing(int64(createAtOffset), offset, false, pageSize)
+				if err != nil {
+					if tries > 3 {
+						mlog.Error("Worker: Failed to get files after multiple retries. Exiting")
+						worker.setJobError(job, model.NewAppError("DoJob", model.NoTranslation, nil, "", http.StatusInternalServerError).Wrap(err))
+						return
+					}
+					mlog.Warn("Failed to get file info for s3 migration. Retrying .. ", mlog.Err(err))
+
+					// Wait a bit before trying again.
+					time.Sleep(15 * time.Second)
+				}
+
+				tries++
+			}
+
+			if len(files) == 0 {
+				mlog.Info("S3PathMigrationWorker: Job is complete", mlog.String("worker", worker.name), mlog.String("job_id", job.Id))
+				worker.setJobSuccess(job)
+				worker.markAsComplete()
+				return
+			}
+
+			// Take batches of `pageSize`
+			for _, f := range files {
+				// We do not fail the job if a single image failed to encode.
+				if f.Path != "" {
+					if err := worker.fileBackend.DecodeFilePathIfNeeded(f.Path); err != nil {
```

Comment:
Maybe include the `f.Id` for additional context if we need to manually patch anything?

Reply:
I didn't get this. Can you clarify where do you want me to add `f.Id`?

Reply:
Hmm, I put this on the wrong line. Basically just proposing logging the `FileInfo.Id` for tracing everywhere.